### ISSUE TITLE
Tensorflow 1.1rc/1.01 too many epochs cause error?

### ISSUE THREAD

**Issue by Windaway** [tensorflow/tensorflow#9067]
Software:Tensorflow 1.1rc Python3.5.27&Tensorflow 1.0.1 Python3.5.3 (Anaconda3) CUDA8.0+Cudnn5.1
OS:Windows 10 X64 1607
HW:i5 6400 Z170 16G 1070

### Describe the problem clearly
When I train my mlp or lenet5, if I choose a large epoch number, then it will get a bad result?

### Source Code / Logs
```
from tensorflow.examples.tutorials.mnist import  input_data
import  tensorflow as tf
mnist=input_data.read_data_sets("MNIST_data/",one_hot=True)
sess=tf.InteractiveSession()
in_units=784
h1_units=300
W1=tf.Variable(tf.truncated_normal([in_units,h1_units],stddev=0.1))
b1=tf.Variable(tf.zeros([h1_units]))
W2=tf.Variable(tf.zeros([h1_units,10]))
b2=tf.Variable(tf.zeros([10]))
x=tf.placeholder(tf.float32,[None,in_units])
keep_prob=tf.placeholder(tf.float32)
hidden1=tf.nn.relu(tf.matmul(x,W1)+b1)
hidden1_drop=tf.nn.dropout(hidden1,keep_prob)
y=tf.nn.softmax(tf.matmul(hidden1_drop,W2)+b2)
y_=tf.placeholder(tf.float32,[None,10])
cross_entropy=tf.reduce_mean(-tf.reduce_sum(y_*tf.log(y),reduction_indices=[1]))
train_step=tf.train.AdagradOptimizer(0.03).minimize(cross_entropy)
tf.global_variables_initializer().run()
correct_prediction=tf.equal(tf.argmax(y,1),tf.argmax(y_,1))
accuracy=tf.reduce_mean(tf.cast(correct_prediction,tf.float32))
for i in range(200000):
    batch_xs,batch_ys=mnist.train.next_batch(500)
    train_step.run({x:batch_xs,y_:batch_ys,keep_prob:0.8})
    if i % 500==0:
        print(i,"--",accuracy.eval({x:batch_xs,y_:batch_ys, keep_prob: 1.0}))

correct_prediction=tf.equal(tf.argmax(y,1),tf.argmax(y_,1))
accuracy=tf.reduce_mean(tf.cast(correct_prediction,tf.float32))
print(accuracy.eval({x:mnist.test.images,y_:mnist.test.labels,keep_prob:1.0}))
```

> C:\Anaconda3\envs\TensorflowRC\python.exe D:/DL/3/mlp.py
> Extracting MNIST_data/train-images-idx3-ubyte.gz
> Extracting MNIST_data/train-labels-idx1-ubyte.gz
> Extracting MNIST_data/t10k-images-idx3-ubyte.gz
> Extracting MNIST_data/t10k-labels-idx1-ubyte.gz
> 2017-04-08 18:07:28.117898: W c:\tf_jenkins\home\workspace\nightly-win\device\gpu\os\windows\tensorflow\core\platform\cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE instructions, but these are available on your machine and could speed up CPU computations.
> 2017-04-08 18:07:28.118165: W c:\tf_jenkins\home\workspace\nightly-win\device\gpu\os\windows\tensorflow\core\platform\cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE2 instructions, but these are available on your machine and could speed up CPU computations.
> 2017-04-08 18:07:28.118511: W c:\tf_jenkins\home\workspace\nightly-win\device\gpu\os\windows\tensorflow\core\platform\cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE3 instructions, but these are available on your machine and could speed up CPU computations.
> 2017-04-08 18:07:28.118745: W c:\tf_jenkins\home\workspace\nightly-win\device\gpu\os\windows\tensorflow\core\platform\cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE4.1 instructions, but these are available on your machine and could speed up CPU computations.
> 2017-04-08 18:07:28.120818: W c:\tf_jenkins\home\workspace\nightly-win\device\gpu\os\windows\tensorflow\core\platform\cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE4.2 instructions, but these are available on your machine and could speed up CPU computations.
> 2017-04-08 18:07:28.121070: W c:\tf_jenkins\home\workspace\nightly-win\device\gpu\os\windows\tensorflow\core\platform\cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use AVX instructions, but these are available on your machine and could speed up CPU computations.
> 2017-04-08 18:07:28.121354: W c:\tf_jenkins\home\workspace\nightly-win\device\gpu\os\windows\tensorflow\core\platform\cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use AVX2 instructions, but these are available on your machine and could speed up CPU computations.
> 2017-04-08 18:07:28.121578: W c:\tf_jenkins\home\workspace\nightly-win\device\gpu\os\windows\tensorflow\core\platform\cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use FMA instructions, but these are available on your machine and could speed up CPU computations.
> 2017-04-08 18:07:28.511257: I c:\tf_jenkins\home\workspace\nightly-win\device\gpu\os\windows\tensorflow\core\common_runtime\gpu\gpu_device.cc:887] Found device 0 with properties: 
> name: GeForce GTX 1070
> major: 6 minor: 1 memoryClockRate (GHz) 1.645
> pciBusID 0000:01:00.0
> Total memory: 8.00GiB
> Free memory: 6.65GiB
> 2017-04-08 18:07:28.511561: I c:\tf_jenkins\home\workspace\nightly-win\device\gpu\os\windows\tensorflow\core\common_runtime\gpu\gpu_device.cc:908] DMA: 0 
> 2017-04-08 18:07:28.511694: I c:\tf_jenkins\home\workspace\nightly-win\device\gpu\os\windows\tensorflow\core\common_runtime\gpu\gpu_device.cc:918] 0:   Y 
> 2017-04-08 18:07:28.511840: I c:\tf_jenkins\home\workspace\nightly-win\device\gpu\os\windows\tensorflow\core\common_runtime\gpu\gpu_device.cc:977] Creating TensorFlow device (/gpu:0) -> (device: 0, name: GeForce GTX 1070, pci bus id: 0000:01:00.0)
> 0 -- 0.404
> 500 -- 0.914
> 1000 -- 0.932
> 1500 -- 0.95
> 2000 -- 0.966
> 2500 -- 0.944
> 3000 -- 0.952
> 3500 -- 0.966

> 
> 80500 -- 1.0
> 81000 -- 1.0
> 81500 -- 1.0
> 82000 -- 1.0
> 82500 -- 0.094
> 83000 -- 0.112
> 83500 -- 0.106
> 84000 -- 0.094
> 84500 -- 0.118
> 85000 -- 0.096
> 85500 -- 0.088
> 86000 -- 0.114
> 86500 -- 0.114
> 87000 -- 0.092
> 87500 -- 0.114
> 88000 -- 0.1
> 88500 -- 0.104

**Comment by Windaway** [comment]
I change the print step to 1. 

> 65972 -- 1.0
> 65973 -- 1.0
> 65974 -- 1.0
> 65975 -- 1.0
> 65976 -- 1.0
> 65977 -- 0.098
> 65978 -- 0.094
> 65979 -- 0.11
> 65980 -- 0.106
> 65981 -- 0.088
> 65982 -- 0.08

The change is suddenly occured.

**Comment by Androbin** [comment]
First, you should initialize W2 the same way as W1 to break the symmetry.
Second, you should compute cross entropy with one of the built-in functions.
It is because of numerical boo boos that you might experience problems.
